File: oop/javaoop/Dog.java
package oop.javaoop ; 

public class Dog extends Animal implements IAnimal {

    private String breed ; 

    public Dog( String name ,  int age , String breed ){
        super(name , age) ; 
        this.breed = breed ; 
    }

    public String getBreed(){
        return this.breed ; 
    }

    public String speak(String sound){
        return sound + " " + "Woo woo" ;
    }
}

File: oop/ts-oop/Dog.ts
import IAnimal from "./Animal.interface";
import { Animal } from "./Animal";

export class Dog extends Animal implements IAnimal {
    breed : string ; 

    constructor(name : string , age : number , breed : string){
        super(name , age)
        this.breed = breed
    }
    
    speak(sound: string): string {
         return sound + " " + "Woo woo" ;
    }
}